hs/src/Reach/Linearize.hs

```diff
@@ -532,6 +573,19 @@ df_eb :: DKExportBlock -> DFApp DLExportBlock
 df_eb (DLinExportBlock at vs b) =
   DLinExportBlock at vs <$> df_bl b
 
+-- Initialize the fluid arrays
+df_init :: DKTail -> DFApp DKTail
+df_init k = do
+  dv <- fixFluidRefType_ =<< DLVar sb Nothing T_Balances <$> df_allocVar
+  nv <- DLVar sb Nothing tokenInfoElemTy <$> df_allocVar
+  let lv = DLV_Let DVC_Many dv
+  let false = DLA_Literal $ DLL_Bool False
+  let zero = DLA_Literal $ DLL_Int sb 0
+  let c1 = DKC_Let sb (DLV_Let DVC_Many nv) $ DLE_LArg sb $ DLLA_Tuple [DLA_Constant DLC_Zero_addr, zero, zero, false]
+  let c2 = DKC_Let sb lv $ DLE_BalanceInit nv
```

Comment:
I don't understand what `DLE_BalanceInit` is for

Reply:
I will remove it as I can just do the conversion to an array directly in this function